homeassistant/components/sensor/yr.py

```diff
@@ -153,50 +151,60 @@ def __init__(self, hass, coordinates, forecast, devices):
         self._url = 'https://aa015h6buqvih86i1.api.met.no/'\
                     'weatherapi/locationforecast/1.9/'
         self._urlparams = coordinates
-        self._nextrun = None
         self._forecast = forecast
         self.devices = devices
         self.data = {}
         self.hass = hass
+        self._unsubscribe_download_new_data = None
+        self._update_time = [randrange(59), randrange(59)]
 
     @asyncio.coroutine
-    def async_update(self, *_):
+    def download_new_data(self, *_):
         """Get the latest data from yr.no."""
         import xmltodict
 
+        if self._unsubscribe_download_new_data:
+            self._unsubscribe_download_new_data()
+            self._unsubscribe_download_new_data = None
+
         def try_again(err: str):
-            """Retry in 15 minutes."""
-            _LOGGER.warning("Retrying in 15 minutes: %s", err)
-            self._nextrun = None
-            nxt = dt_util.utcnow() + timedelta(minutes=15)
-            if nxt.minute >= 15:
-                async_track_point_in_utc_time(self.hass, self.async_update,
-                                              nxt)
-
-        if self._nextrun is None or dt_util.utcnow() >= self._nextrun:
-            try:
-                websession = async_get_clientsession(self.hass)
-                with async_timeout.timeout(10, loop=self.hass.loop):
-                    resp = yield from websession.get(
-                        self._url, params=self._urlparams)
-                if resp.status != 200:
-                    try_again('{} returned {}'.format(resp.url, resp.status))
-                    return
-                text = yield from resp.text()
-
-            except (asyncio.TimeoutError, aiohttp.ClientError) as err:
-                try_again(err)
+            """Retry in at least 20 minutes."""
+            minutes = 15 + randrange(5)
```

Comment:
This is not 'at least 20 minutes'. This is '15 to 20 minutes'.

Reply:
Yes, I will update the comment